yt_dlp/extractor/common.py

```diff
@@ -847,7 +847,7 @@ def __can_accept_status_code(err, expected_status):
         else:
             return err.status in variadic(expected_status)
 
-    def _create_request(self, url_or_request, data=None, headers=None, query=None, extensions=None):
+    def _create_request(self, url_or_request, data=None, headers=None, query=None, extensions=None, allow_redirects=True):
```

Comment:
We want to avoid adding parameters that are rarely used to these functions, so I would instead in the caller create a Request object with the allow_redirects extension set.